test/labeling/model/test_label_model.py

```diff
@@ -17,36 +16,40 @@ def setUpClass(cls):
         cls.m = 10
         cls.k = 2
 
-    def _test_label_model(self, data, test_acc=True):
-        label_model = LabelModel(k=data.k, verbose=False)
-        label_model.train_model(
-            data.L,
-            deps=data.E,
-            class_balance=data.p,
-            n_epochs=1000,
-            log_train_every=200,
+    def test_model_init(self):
+        label_model = LabelModel(k=2, verbose=False)
+
+        # Test class balance
+        Y_dev = np.array([1, 1, 2, 2, 1, 1, 1, 1, 2, 2])
+        label_model._set_class_balance(class_balance=None, Y_dev=Y_dev)
+        self.assertTrue((label_model.p == np.array([0.6, 0.4])).all())
+
+        # Test dimension constants
```

Comment:
Why is this in the same test? Do we need the class balance set already?

Reply:
good catch, separating